src/plugins/explore/public/components/query_panel/components/footer/show_field.tsx

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState } from 'react';
+import { EuiButtonEmpty } from '@elastic/eui';
+import { i18n } from '@osd/i18n';
+
+interface ShowFieldToggleProps {
+  isEnabled: boolean;
+  onToggle: (enabled: boolean) => void;
+}
+
+export const ShowFieldToggle: React.FC<ShowFieldToggleProps> = ({ isEnabled, onToggle }) => {
+  const [showField, setShowField] = useState(isEnabled);
```

Comment:
seems isEnabled should sync with show field, and show field looks to be a duplicated state, is this state necessary?